neurots/morphmath/sample.py

```diff
@@ -145,3 +145,13 @@ def ph(phs, random_generator=np.random):
     """Samples randomly a persistence diagram from the input distribution."""
     index = random_generator.choice(len(phs))
     return phs[index]
+
+
+def sample_spherical_unit_vectors(rng):
+    """Sample a point uniformly on the sphere.
+
+    Args:
+        rng: random number generator
+    """
+    x = rng.normal(0, 1, 3)
```

Comment:
I *think* this is wrong. If you want to uniformly sample on a sphere you need to sample the angles, not x-y-z. Unless what you need to do here is not the same as randomly sample one point on a sphere :) please clarify, as this is an important point

Reply:
Not for the code, but for the logic of the issue is you sample x,y,z instead: https://www.bogotobogo.com/Algorithms/uniform_distribution_sphere.php

Reply:
what is wrong exactly? I want x, y, z on a sphere, uniformly sampled, which is achieved by this 'trick'. It's faster than sampling angles and computing trigonometrics functions.

Reply:
eq (16) here: https://mathworld.wolfram.com/SpherePointPicking.html

Reply:
let's talk in person, this is a hard concept but it's not correct to compute this way (I'll send plot in slack)

Reply:
got it, works this way as well. maybe I should replace the other implementation :+1: